enforcer-parent/enforcer/src/test/java/org/wso2/choreo/connect/enforcer/backend/jwt/BackendJwtTest.java

```diff
@@ -0,0 +1,98 @@
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.enforcer.backend.jwt;
+
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.RSASSAVerifier;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.util.X509CertUtils;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.wso2.carbon.apimgt.common.gateway.dto.JWTConfigurationDto;
+import org.wso2.carbon.apimgt.common.gateway.dto.JWTInfoDto;
+import org.wso2.carbon.apimgt.common.gateway.dto.JWTValidationInfo;
+import org.wso2.carbon.apimgt.common.gateway.jwtgenerator.APIMgtGatewayJWTGeneratorImpl;
+import org.wso2.carbon.apimgt.common.gateway.jwtgenerator.AbstractAPIMgtGatewayJWTGenerator;
+import org.wso2.choreo.connect.enforcer.commons.exception.EnforcerException;
+import org.wso2.choreo.connect.enforcer.util.JWTUtils;
+import org.wso2.choreo.connect.enforcer.util.TLSUtils;
+import org.wso2.choreo.connect.enforcer.util.TLSUtilsTest;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPublicKey;
+
+public class BackendJwtTest {
+    private static final String keystore = "keystore";
+    private static final String publicCert = "mg.pem";
+    private static final String privateKey = "mg.key";
+
+    private static final String certPath = BackendJwtTest.class.getProtectionDomain().getCodeSource().
+            getLocation().getPath() + keystore + File.separator + publicCert;
+    private static final String keyPath = BackendJwtTest.class.getProtectionDomain().getCodeSource().
+            getLocation().getPath() + keystore + File.separator + privateKey;
+    private static JWTConfigurationDto jwtConfig;
+    private static AbstractAPIMgtGatewayJWTGenerator jwtGenerator;
+    // Init JWT generator
+    @BeforeClass
+    public static void start() throws Exception {
+        jwtConfig = getConfigDto();
+        jwtGenerator = new APIMgtGatewayJWTGeneratorImpl();
+        jwtGenerator.setJWTConfigurationDto((jwtConfig));
+    }
+
+    public static JWTConfigurationDto getConfigDto() {
+        JWTConfigurationDto configDto = new JWTConfigurationDto();
+        try {
+            configDto.setPublicCert(TLSUtils.getCertificateFromFile(certPath));
+            configDto.setPrivateKey(JWTUtils.getPrivateKey(keyPath));
+        } catch (EnforcerException | CertificateException | IOException e) {
+            throw new RuntimeException(e);
+        }
+        configDto.setEnabled(true);
+        configDto.setEnableUserClaims(false);
+        configDto.setGatewayJWTGeneratorImpl("org.wso2.carbon.apimgt.common.gateway.jwtgenerator.APIMgtGatewayJWTGeneratorImpl");
+        configDto.setTtl(3600);
+        return configDto;
+    }
+
```

Comment:
If possible let's add descriptions to the test cases. It will help to fix when there's a test case failure later.
Please consider other tests in this PR as well.

Reply:
understood

Reply:
Since this is an enforcer test case, adding a comment above a test case is enough. (Due to the unavailability of testng)